bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsBridgeHandler.java

```diff
@@ -0,0 +1,880 @@
+    private synchronized void sendElroMessage(DatagramSocket socket, ElroConnectsMessage elroMessage,
+            boolean waitResponse) throws IOException {
+        String message = gsonOut.toJson(elroMessage);
+        awaitResponse(waitResponse);
+        send(socket, message, false);
+    }
+
+    /**
+     * Send device control command. The device command string various by device type. The calling method is responsible
+     * for creating the appropriate command string.
+     *
+     * @param deviceId
+     * @param deviceCommand Elro Connects device command string
+     * @throws IOException
+     */
+    public void deviceControl(int deviceId, String deviceCommand) throws IOException {
+        String connectorId = this.connectorId;
+        String ctrlKey = this.ctrlKey;
+        DatagramSocket socket = this.socket;
+        if (!connectorId.isEmpty() && !ctrlKey.isEmpty() && (socket != null)) {
+            logger.debug("Device control {}, status {}", deviceId, deviceCommand);
+            ElroConnectsMessage elroMessage = new ElroConnectsMessage(msgIdIncrement(), connectorId, ctrlKey,
+                    ELRO_DEVICE_CONTROL).withDeviceId(ElroConnectsUtil.encode(deviceId))
+                            .withDeviceStatus(deviceCommand);
+            sendElroMessage(socket, elroMessage, false);
+        } else {
+            // This should not happen if properly initialized
+            logger.debug("Could not send");
+        }
+    }
+
+    /**
+     * Send request to receive all device names.
+     *
+     * @throws IOException
+     */
+    private void getDeviceNames() throws IOException {
+        String connectorId = this.connectorId;
+        String ctrlKey = this.ctrlKey;
+        DatagramSocket socket = this.socket;
+        if (!connectorId.isEmpty() && !ctrlKey.isEmpty() && (socket != null)) {
+            logger.debug("Get device names");
+            ElroConnectsMessage elroMessage = new ElroConnectsMessage(msgIdIncrement(), connectorId, ctrlKey,
+                    ELRO_GET_DEVICE_NAME).withDeviceId(0);
+            sendElroMessage(socket, elroMessage, true);
+        } else {
+            // This should not happen if properly initialized
+            logger.debug("Could not send");
+        }
+    }
+
+    /**
+     * Send request to receive all device statuses.
+     *
+     * @throws IOException
+     */
+    private void getDeviceStatuses() throws IOException {
+        String connectorId = this.connectorId;
+        String ctrlKey = this.ctrlKey;
+        DatagramSocket socket = this.socket;
+        if (!connectorId.isEmpty() && !ctrlKey.isEmpty() && (socket != null)) {
+            logger.debug("Get all equipment status");
+            ElroConnectsMessage elroMessage = new ElroConnectsMessage(msgIdIncrement(), connectorId, ctrlKey,
+                    ELRO_GET_DEVICE_STATUSES);
+            sendElroMessage(socket, elroMessage, true);
+        } else {
+            // This should not happen if properly initialized
```

Comment:
As above, since this is occurring multiple times, it may be worth looking at the suggestion to make the code shorter.